### PR TITLE
Support esm cjs and split demo app into own project

### PR DESCRIPTION
**Pull request by markwylde**
- Implements linting via the airbnb-typescript guidelines
- Implements cjs/esm conditional imports
- abstracts demo into own bundles project while keeping library pure tsc
- add multiple version node CI/CD pipeline

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
 
-      - name: Run Tests 🧪
-        run: |
-          npm run test
+    - run: npm install
+    - run: npm run build --if-present
```
Comment (votemike) [inline .github/workflows/test.yml:27]: Do you need to build?

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,6 @@
 node_modules
-dist/*.min.js
+demo/dist/*.js
```
Comment (votemike) [inline .gitignore:2]: demo/dist/*?

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,6 @@
...
+demo/dist/*.map.js
+dist/*.js
```
Comment (votemike) [inline .gitignore:4]: dist/*?

File: demo/package.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "react-accordion-demo",
+  "version": "1.0.0",
+  "description": "A really lightweight, symantic, super simple accordion for React",
+  "type": "module",
+  "scripts": {
+    "develop": "servatron -d dist & esbuild --watch --bundle src/index.tsx --outfile=dist/index.min.js --loader:.js=jsx",
```
Comment (votemike) [inline demo/package.json:7]: Does this watch both the demo code AND the accordion code?

File: demo/package.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/votemike/react-accordion/issues"
+  },
+  "homepage": "https://github.com/votemike/react-accordion#readme",
+  "dependencies": {
+    "react": "^18.2.0",
+    "react-accordion": "../"
+  },
+  "devDependencies": {
```
Comment (votemike) [inline demo/package.json:30]: When does the demo have dev dependencies?

File: demo/src/index.tsx [hunk 6]
```diff
@@ -24,8 +24,13 @@ const DemoPage = () => {
...
+    console.error('Could not find the <app> element on the DOM');
+    return;
+  }
+
   const root = ReactDOM.createRoot(appNode);
   root.render(<DemoPage />);
 }
 
-document.addEventListener('DOMContentLoaded', main);
+document.addEventListener('DOMContentLoaded', main);
```
Comment (votemike) [inline demo/src/index.tsx:36]: Why are you removing new lines 😢

File: dist/cjs/index.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+"use strict";
```
Comment (votemike) [inline dist/cjs/index.js:1]: Should this file be ignored?

File: dist/cjs/index.js.map [hunk 8]
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"index.js","sourceRoot":"","sources":["../../src/index.tsx"],"names":[],"mappings":";;;AAWA,SAAS,cAAc,CAAC,KAA0B;IAChD,MAAM,EAAE,KAAK,EAAE,GAAG,KAAK,CAAC;IAExB,OAAO,CACL,yCACG,KAAK,CAAC,GAAG,CAAC,CAAC,IAAI,EAAE,EAAE,CAAC,CACnB,yCACE,iEAAqB,IAAI,CAAC,IAAI,EAAE,IAAI,EAAE,IAAI,CAAC,IAAI,iBAC7C,8CAAU,IAAI,CAAC,OAAO,GAAW,EAChC,IAAI,CAAC,KAAK,KACH,IAJH,IAAI,CAAC,IAAI,CAKb,CACN,CAAC,GACC,CACN,CAAC;AACJ,CAAC;AAED,kBAAe,cAAc,CAAC"}
```
Comment (votemike) [inline dist/cjs/index.js.map:1]: Should this file be ignored?

File: dist/esm/index.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
```
Comment (votemike) [inline dist/esm/index.js:1]: Should this file be ignored?

File: dist/esm/index.js.map [hunk 10]
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"index.js","sourceRoot":"","sources":["../../src/index.tsx"],"names":[],"mappings":";AAWA,SAAS,cAAc,CAAC,KAA0B;IAChD,MAAM,EAAE,KAAK,EAAE,GAAG,KAAK,CAAC;IAExB,OAAO,CACL,uBACG,KAAK,CAAC,GAAG,CAAC,CAAC,IAAI,EAAE,EAAE,CAAC,CACnB,uBACE,iCAAqB,IAAI,CAAC,IAAI,EAAE,IAAI,EAAE,IAAI,CAAC,IAAI,aAC7C,4BAAU,IAAI,CAAC,OAAO,GAAW,EAChC,IAAI,CAAC,KAAK,IACH,IAJH,IAAI,CAAC,IAAI,CAKb,CACN,CAAC,GACC,CACN,CAAC;AACJ,CAAC;AAED,eAAe,cAAc,CAAC"}
```
Comment (votemike) [inline dist/esm/index.js.map:1]: Should this file be ignored?

File: dist/types/index.d.ts [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import { ReactNode } from 'react';
```
Comment (votemike) [inline dist/types/index.d.ts:1]: Should this file be ignored?

File: .github/workflows/test.yml (hunk 1 above)
Comment (markwylde) [reply .github/workflows/test.yml]: Yup, but we don't need the `--if-present`. We want the CI/CD to rest the typescript builds. Wanting a reason I love not using typescript? No built needed! :P

Fixed in 676c4cd1bbacd250584b55551d7bdb69674978bc.

File: demo/package.json (hunk 4 above)
Comment (markwylde) [reply demo/package.json:7]: Nope. Demo would be a separate self contained project.

File: demo/package.json (hunk 5 above)
Comment (markwylde) [reply demo/package.json:30]: Not sure I understand the question. You can see them below?

File: .gitignore (hunk 2 above)
Comment (markwylde) [reply .gitignore:2]: Can't be unfortunatley as we need the `index.html`, but later on we could copy it from the src directory at build.

File: demo/package.json (hunk 5 above)
Comment (votemike) [reply demo/package.json:30]: Is the demo just for use locally? If so, why do any of the dependencies need to be dev?

Comment (markwylde) [reply demo/package.json:30]: Think of `demo` as it's own [packgage](https://docs.npmjs.com/cli/v7/using-npm/workspaces) or project in itself. The demo should also be a reflection of a project using the react-accordion, of which you would expect the dependencies to be separated into their proper scope.

**Comment by votemike** [comment]
I will review once merge conflicts are resolved.

File: .eslintrc [hunk 12]
```diff
@@ -1,3 +1,13 @@
...
+  "root": true,
+  "extends": [
+    "airbnb",
+    "airbnb-typescript"
+  ],
+  "parserOptions": {
+    "project": "./tsconfig.json"
+  },
+  "rules": {
+    "import/no-import-module-exports": "off"
```
Comment (markwylde) [inline .eslintrc:11]: Bug logged with eslint [here](https://github.com/import-js/eslint-plugin-import/issues/2181).

**Comment by votemike** [comment]
When running `npm install`, my package-lock.json seems to update

**Comment by cat-markwylde** [comment]
> When running npm install, my package-lock.json seems to update.
> Both at the root and within the demo folder.

Weird. Should be [fixed](https://github.com/votemike/react-accordion/pull/1/commits/ac871348ae1ff2f46be2d19cf43e7304816a07a9) now.

**Review by votemike — CHANGES_REQUESTED** [review]
package-lock.json changes upon npm install in the demo directory.

**Review by votemike — APPROVED** [review]

**Pull request merged**